test/inductor/test_torchinductor_strided_blocks.py

```diff
@@ -1003,6 +1004,65 @@ def test_pointwise_index_order(self):
     xindex = xoffset + tl.arange(0, XBLOCK)[None, None, :]""",  # noqa: B950
         )
 
+    @config.patch("triton.prefer_nd_tiling", True)
+    @config.patch("triton.max_tiles", 3)
+    @parametrize(
+        "block_multiple, ynumel_exceed_ygrid_size",
+        [
+            # no xdim boundary check, ydim is checked
+            [True, True],
+            # no boundary check in both dimensions
+            [True, False],
+            # if numel not a block multiple, must boundary check
+            # skip triton_cpu very slow test > 1000s
+            subtest([False, False], decorators=[test_torchinductor.skip_if_triton_cpu])
+            # TODO: test zdim too
+        ],
+    )
+    def test_boundary_check(self, block_multiple, ynumel_exceed_ygrid_size):
+        from torch._inductor.runtime.hints import TRITON_MAX_BLOCK
+        from torch._inductor.runtime.runtime_utils import get_max_y_grid
+
+        shape = [TRITON_MAX_BLOCK["Y"], TRITON_MAX_BLOCK["X"]]
+
+        if not block_multiple:
+            shape = [s + 1 for s in shape]
+
+        if ynumel_exceed_ygrid_size:
+            shape[0] = shape[0] * (math.ceil(get_max_y_grid() / shape[0])) + shape[0]
+
+        a = torch.zeros(shape, device=self.device)
+        b = torch.zeros((shape[0], 1), device=self.device)
+
+        def func(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+            return a + b
+
+        result, code = run_and_compare(
+            self,
+            func,
+            a,
+            b,
+            expected_num_triton_kernels=1,
+            expected_num_block_pointers=3,
+        )
+
+        code = code[0]
+        if block_multiple:
+            if ynumel_exceed_ygrid_size:
+                # Only the y dimension should be boundary checked
+                # Loading a
+                self.assertTrue("boundary_check=[1]" in code)
```

Comment:
very nit: prefer `self.assertIn`
```suggestion
                self.assertIn("boundary_check=[1]", code)
```